// tmdb-ts/dist/api.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Api = void 0;
// Using native fetch API (available in Node 18+ and all modern browsers)
const utils_1 = require("./utils");
const constants_1 = require("./common/constants");
class Api {
    constructor(accessToken, customFetch) {
        this.accessToken = accessToken;
        this.accessToken = accessToken;
        this.fetchFn = customFetch ?? fetch;
    }
    /* eslint-disable  @typescript-eslint/no-explicit-any */
    async get(path, options) {
        const params = (0, utils_1.parseOptions)(options);
        const response = await this.fetchFn(`${constants_1.BASE_URL_V3}${path}?${params}`, {
            method: 'GET',
            headers: {
                Authorization: `Bearer ${this.accessToken}`,
                'Content-Type': 'application/json;charset=utf-8',
            },
        });
        if (!response.ok) {
            return Promise.reject((await response.json()));
        }
        return (await response.json());
    }
}
exports.Api = Api;
