// tmdb-ts/dist/tmdb.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TMDB = void 0;
const endpoints_1 = require("./endpoints");
const companies_1 = require("./endpoints/companies");
const networks_1 = require("./endpoints/networks");
class TMDB {
    constructor(accessToken, options) {
        this.accessToken = accessToken;
        this.customFetch = options?.fetch;
    }
    get account() {
        return new endpoints_1.AccountEndpoint(this.accessToken, this.customFetch);
    }
    get configuration() {
        return new endpoints_1.ConfigurationEndpoint(this.accessToken, this.customFetch);
    }
    get certifications() {
        return new endpoints_1.CertificationEndpoint(this.accessToken, this.customFetch);
    }
    get changes() {
        return new endpoints_1.ChangeEndpoint(this.accessToken, this.customFetch);
    }
    get credits() {
        return new endpoints_1.CreditsEndpoint(this.accessToken, this.customFetch);
    }
    get companies() {
        return new companies_1.CompaniesEndpoint(this.accessToken, this.customFetch);
    }
    get networks() {
        return new networks_1.NetworksEndpoint(this.accessToken, this.customFetch);
    }
    get search() {
        return new endpoints_1.SearchEndpoint(this.accessToken, this.customFetch);
    }
    get genres() {
        return new endpoints_1.GenreEndpoint(this.accessToken, this.customFetch);
    }
    get movies() {
        return new endpoints_1.MoviesEndpoint(this.accessToken, this.customFetch);
    }
    get tvShows() {
        return new endpoints_1.TvShowsEndpoint(this.accessToken, this.customFetch);
    }
    get tvEpisode() {
        return new endpoints_1.TvEpisodesEndpoint(this.accessToken, this.customFetch);
    }
    get discover() {
        return new endpoints_1.DiscoverEndpoint(this.accessToken, this.customFetch);
    }
    get people() {
        return new endpoints_1.PeopleEndpoint(this.accessToken, this.customFetch);
    }
    get review() {
        return new endpoints_1.ReviewEndpoint(this.accessToken, this.customFetch);
    }
    get trending() {
        return new endpoints_1.TrendingEndpoint(this.accessToken, this.customFetch);
    }
    get find() {
        return new endpoints_1.FindEndpoint(this.accessToken, this.customFetch);
    }
    get keywords() {
        return new endpoints_1.KeywordsEndpoint(this.accessToken, this.customFetch);
    }
    get collections() {
        return new endpoints_1.CollectionsEndpoint(this.accessToken, this.customFetch);
    }
    get tvSeasons() {
        return new endpoints_1.TvSeasonsEndpoint(this.accessToken, this.customFetch);
    }
    get watchProviders() {
        return new endpoints_1.WatchProvidersEndpoint(this.accessToken, this.customFetch);
    }
}
exports.TMDB = TMDB;
